### ISSUE TITLE
New Search UI doesnt honour SEARCH_PAGE ACL

### ISSUE THREAD

**Issue by SammyIsConfused** [openequella/openEQUELLA#2727]
**Describe the bug**
Even though it doesn't show up in the left hand menu, if a user does not have the SEARCH_PAGE ACL, they can still route to the new search page and use it.

**To Reproduce**

1. Set up a user without SEARCH_PAGE granted.
2. Route to {institution URL}/access/search
3. Note that the search page is served up and is usable.

**Expected behavior**
If SEARCH_PAGE is revoked, a user should not be able to use the search page.

**Platform:**

- OpenEquella Version: 2020.2 onwards
- OS: Agnostic
- Browser: Agnostic

**Comment by SammyIsConfused** [comment]
After looking into this, it appears that the new search is already protected by DISCOVER_ITEM. If this is revoked then all results will be empty from searches, in the old UI, new UI and using the APIs as standalone. 

Because of that, the SEARCH_PAGE ACL seems redundant - all it controls is whether Search appears in the menu bar. In the old UI it blocks access to the search page even if you route to it directly - but in the new UI since we don’t do server side rendering, it just takes you to the search page. Administrators should rely on DISCOVER_ITEM to protect searching, not SEARCH_PAGE. No code changes are necessary to support this.

**To summarise:**
_Old UI - SEARCH_PAGE_ protects access to the search page (throws a full page 403 if you route to it) and hides the Search option from the menu bar.

_New UI  - SEARCH_PAGE_ hides the Search option from the menu bar. Routing to the search page directly still works. 

_Old UI & New UI and all search APIs_ - DISCOVER_ITEM  protects searches themselves. If revoked, the search will return no items. SEARCH_PAGE by default seems to be granted to Logged In Users. DISCOVER_ITEM by default seems to be granted to Content Administrator Role.
